### DISCUSSION TITLE
Set focus in Elmish mode

### DISCUSSION THREAD

**Discussion by pth65** [fsprojects/Avalonia.FuncUI#442] [Q&A]
Hi,

I would like to toggle the focus between a TextBox and a ListBox to set which control will get keyboard input.
I am writing a small utility for personal use where one of the features would be to navigate folders and files with the keyboard, using vi key shortcuts.
My idea is to have a hotkey (only for my app) that switch focus between the 2 controls.
I saw there is an example setting focus but not for Elmish mode.
Any help would be appreciated, thank you.

**Comment by JaggerJo** [comment]
Hey,

Working with focus in Elmish is a bit problematic as you need to have a reference to the control you want to focus. You could work around this by using `FindControl` by name to find the control you want to focus.

https://reference.avaloniaui.net/api/Avalonia.Controls/ControlExtensions/D49E0C29

https://github.com/fsprojects/Avalonia.FuncUI/blob/master/src/Examples/Examples.SettingFocus/Program.fs

**Comment by pth65** [comment]
Hi,

Thank you for your quick answer.
I am not sure to understand how to get the underlying component from an IView.
I built the following view:

```
let mainView state dispatch =
        Grid.create [
…
            Grid.children [
                // Search
		TextBox.create [
		        TextBox.name "SearchTextBox"
…
                // Dir / Files
                ListBox.create [
                    ListBox.name "DirFilesListBox"
…
                ]
            ]
        ]

```

I tried to get the underlying component from my mainView with something like here below:

```
let private viewComponent view = Component (fun ctx ->
        view
    )
```

where the view is my mainView (IView<Grid>) but the FindControl throws an exception because it didn't find the component.

**Comment by toubou-k** [comment]
This is quite an old thread, but in case someone runs into this issue:
I found that using View.withOutlet allows access to the control instance and solves this kind of problem.
I also came across a helpful article explaining it in more detail: https://zenn.dev/d3/articles/90d0769b007b74
